### ISSUE TITLE
subject area client broken (regression)

### ISSUE THREAD

**Issue by planetf1** [odpi/egeria#3257]
```
jonesn:open-metadata-samples/ (issue3254b) $ java -jar ./access-services-samples/subject-area-client-samples/subject-area-definition-sample/target/subject-area-definition-sample-2.0-jar-with-dependencies.jar
Enter the OMAG Server platform URL. Press enter to get the default (https://localhost:9443)):
Enter a OMAG Server Name. Press enter to get the default (cocoMDS4)):
Enter a userId. Press enter to get the default (erinoverview)):
===============================
Subject Area Definition Sample
===============================
Running against OMAG server platform: https://localhost:9443
OMAG server platform: cocoMDS4
Using userId: erinoverview
----------------------------
Creating the Coco Pharmaceutical Glossary for Subject Area Definitions :
Exception: java.lang.IllegalArgumentException with message can't parse argument number: "timestamp":"2020-06-25T11:28:26.818+00:00"
jonesn:open-metadata-samples/ (issue3254b) $
```

The exception comes back from the Spring Rest Connector after SubjectAreaBaseImpl:postRESTCall where we get:
```
RESTServerException{reportedHTTPCode=503, reportingClassName='org.odpi.openmetadata.adapters.connectors.restclients.spring.SpringRESTClientConnector', reportingActionDescription='createGlossary', errorMessage='CLIENT-SIDE-REST-API-CONNECTOR-503-002 A client-side exception org.springframework.web.client.HttpServerErrorException$InternalServerError was received by method createGlossary from API call https://localhost:9443/servers/cocoMDS4/open-metadata/access-services/subject-area/users/erinoverview/glossaries to server cocoMDS4 on platform https://localhost:9443.  The error message was 500 : [{"timestamp":"2020-06-25T11:46:45.139+00:00","status":500,"error":"Internal Server Error","message":"","path":"/servers/cocoMDS4/open-metadata/access-services/subject-area/users/erinoverview/glossaries"}]', reportedSystemAction='The client has issued a call to the open metadata access service REST API in a remote server and has received an exception from the local client libraries.', reportedUserAction='Review the error message to determine the cause of the error.  Check that the server is running an the URL is correct. Look for errors in the local server's console to understand and correct the cause of the error. Then rerun the request', reportedCaughtException=org.springframework.web.client.HttpServerErrorException$InternalServerError: 500 : [{"timestamp":"2020-06-25T11:46:45.139+00:00","status":500,"error":"Internal Server Error","message":"","path":"/servers/cocoMDS4/open-metadata/access-services/subject-area/users/erinoverview/glossaries"}]}
```

I know there was refactoring to use this connector made recently as well as other changes in SA.

@davidradl can you check you're comfortable this isn't a blocking issue for release 2.0? Subject area is still marked as 'in development'

**Comment by planetf1** [comment]
Subject Area FVT also fails:
```
/Library/Java/JavaVirtualMachines/adoptopenjdk-14-openj9.jdk/Contents/Home/bin/java -javaagent:/Users/jonesn/Library/Application Support/JetBrains/Toolbox/apps/IDEA-U/ch-0/202.5792.28/IntelliJ IDEA 2020.2 EAP.app/Contents/lib/idea_rt.jar=59862:/Users/jonesn/Library/Application Support/JetBrains/Toolbox/apps/IDEA-U/ch-0/202.5792.28/IntelliJ IDEA 2020.2 EAP.app/Contents/bin -Dfile.encoding=UTF-8 -classpath /Users/jonesn/IdeaProjects/egeria/open-metadata-test/open-metadata-fvt/access-services-fvt/subject-area-fvt/target/classes:/Users/jonesn/IdeaProjects/egeria/open-metadata-implementation/access-services/subject-area/subject-area-api/target/classes:/Users/jonesn/.m2/repository/com/fasterxml/jackson/core/jackson-databind/2.11.0/jackson-databind-2.11.0.jar:/Users/jonesn/.m2/repository/com/fasterxml/jackson/core/jackson-core/2.11.0/jackson-core-2.11.0.jar:/Users/jonesn/IdeaProjects/egeria/open-metadata-implementation/repository-services/repository-services-apis/target/classes:/Users/jonesn/.m2/repository/org/apache/commons/commons-collections4/4.4/commons-collections4-4.4.jar:/Users/jonesn/IdeaProjects/egeria/open-metadata-implementation/frameworks/open-connector-framework/target/classes:/Users/jonesn/.m2/repository/com/fasterxml/jackson/core/jackson-annotations/2.11.0/jackson-annotations-2.11.0.jar:/Users/jonesn/IdeaProjects/egeria/open-metadata-implementation/common-services/ffdc-services/target/classes:/Users/jonesn/IdeaProjects/egeria/open-metadata-implementation/adapters/open-connectors/rest-client-connectors/rest-client-connectors-api/target/classes:/Users/jonesn/IdeaProjects/egeria/open-metadata-implementation/adapters/open-connectors/rest-client-connectors/rest-client-factory/target/classes:/Users/jonesn/IdeaProjects/egeria/open-metadata-implementation/adapters/open-connectors/rest-client-connectors/spring-rest-client-connector/target/classes:/Users/jonesn/.m2/repository/org/springframework/spring-web/5.2.7.RELEASE/spring-web-5.2.7.RELEASE.jar:/Users/jonesn/.m2/repository/org/springframework/spring-beans/5.2.7.RELEASE/spring-beans-5.2.7.RELEASE.jar:/Users/jonesn/.m2/repository/org/springframework/spring-core/5.2.7.RELEASE/spring-core-5.2.7.RELEASE.jar:/Users/jonesn/.m2/repository/org/springframework/spring-jcl/5.2.7.RELEASE/spring-jcl-5.2.7.RELEASE.jar:/Users/jonesn/.m2/repository/org/codehaus/plexus/plexus-utils/3.3.0/plexus-utils-3.3.0.jar:/Users/jonesn/.m2/repository/org/apache/commons/commons-lang3/3.10/commons-lang3-3.10.jar:/Users/jonesn/.m2/repository/org/slf4j/slf4j-api/1.7.30/slf4j-api-1.7.30.jar:/Users/jonesn/IdeaProjects/egeria/open-metadata-implementation/frameworks/audit-log-framework/target/classes:/Users/jonesn/IdeaProjects/egeria/open-metadata-implementation/access-services/subject-area/subject-area-client/target/classes org.odpi.openmetadata.accessservices.subjectarea.fvt.RunAllFVTNow
Enter a URL. Press enter to get the default (https://localhost:9443.)) :
Enter a OMAG Server Name. Press enter to get the default (cocoMDS4)) :
Enter a userId. Press enter to get the default (erinoverview)) :
SLF4J: Failed to load class "org.slf4j.impl.StaticLoggerBinder".
SLF4J: Defaulting to no-operation (NOP) logger implementation
SLF4J: See http://www.slf4j.org/codes.html#StaticLoggerBinder for further details.
Exception in thread "main" java.lang.IllegalArgumentException: can't parse argument number: "timestamp":"2020-06-25T12:02:14.204+00:00"
	at java.base/java.text.MessageFormat.makeFormat(MessageFormat.java:1454)
	at java.base/java.text.MessageFormat.applyPattern(MessageFormat.java:492)
	at java.base/java.text.MessageFormat.<init>(MessageFormat.java:371)
	at org.odpi.openmetadata.frameworks.auditlog.MessageFormatter.formatMessage(MessageFormatter.java:38)
	at org.odpi.openmetadata.frameworks.auditlog.MessageFormatter.getFormattedMessage(MessageFormatter.java:92)
	at org.odpi.openmetadata.frameworks.connectors.ffdc.OCFCheckedExceptionBase.<init>(OCFCheckedExceptionBase.java:75)
	at org.odpi.openmetadata.frameworks.connectors.ffdc.OCFCheckedExceptionBase.<init>(OCFCheckedExceptionBase.java:57)
	at org.odpi.openmetadata.commonservices.ffdc.exceptions.OMAGCheckedExceptionBase.<init>(OMAGCheckedExceptionBase.java:29)
	at org.odpi.openmetadata.accessservices.subjectarea.ffdc.exceptions.SubjectAreaCheckedException.<init>(SubjectAreaCheckedException.java:32)
	at org.odpi.openmetadata.accessservices.subjectarea.ffdc.exceptions.PropertyServerException.<init>(PropertyServerException.java:23)
	at org.odpi.openmetadata.accessservices.subjectarea.client.SubjectAreaBaseImpl.findRESTCall(SubjectAreaBaseImpl.java:260)
	at org.odpi.openmetadata.accessservices.subjectarea.client.SubjectAreaGlossaryImpl.findGlossary(SubjectAreaGlossaryImpl.java:132)
	at org.odpi.openmetadata.accessservices.subjectarea.fvt.GlossaryFVT.findGlossaries(GlossaryFVT.java:210)
	at org.odpi.openmetadata.accessservices.subjectarea.fvt.GlossaryFVT.run(GlossaryFVT.java:61)
	at org.odpi.openmetadata.accessservices.subjectarea.fvt.GlossaryFVT.runIt(GlossaryFVT.java:57)
	at org.odpi.openmetadata.accessservices.subjectarea.fvt.RunAllFVTNow.main(RunAllFVTNow.java:27)
Caused by: java.lang.NumberFormatException: For input string: ""timestamp":"2020-06-25T12:02:14.204+00:00""
	at java.base/java.lang.NumberFormatException.forInputString(NumberFormatException.java:68)
	at java.base/java.lang.Integer.parseInt(Integer.java:638)
	at java.base/java.lang.Integer.parseInt(Integer.java:770)
	at java.base/java.text.MessageFormat.makeFormat(MessageFormat.java:1452)
	... 15 more

Process finished with exit code 1
```
And also the 2-server version
```
/Library/Java/JavaVirtualMachines/adoptopenjdk-14-openj9.jdk/Contents/Home/bin/java -javaagent:/Users/jonesn/Library/Application Support/JetBrains/Toolbox/apps/IDEA-U/ch-0/202.5792.28/IntelliJ IDEA 2020.2 EAP.app/Contents/lib/idea_rt.jar=59884:/Users/jonesn/Library/Application Support/JetBrains/Toolbox/apps/IDEA-U/ch-0/202.5792.28/IntelliJ IDEA 2020.2 EAP.app/Contents/bin -Dfile.encoding=UTF-8 -classpath /Users/jonesn/IdeaProjects/egeria/open-metadata-test/open-metadata-fvt/access-services-fvt/subject-area-fvt/target/classes:/Users/jonesn/IdeaProjects/egeria/open-metadata-implementation/access-services/subject-area/subject-area-api/target/classes:/Users/jonesn/.m2/repository/com/fasterxml/jackson/core/jackson-databind/2.11.0/jackson-databind-2.11.0.jar:/Users/jonesn/.m2/repository/com/fasterxml/jackson/core/jackson-core/2.11.0/jackson-core-2.11.0.jar:/Users/jonesn/IdeaProjects/egeria/open-metadata-implementation/repository-services/repository-services-apis/target/classes:/Users/jonesn/.m2/repository/org/apache/commons/commons-collections4/4.4/commons-collections4-4.4.jar:/Users/jonesn/IdeaProjects/egeria/open-metadata-implementation/frameworks/open-connector-framework/target/classes:/Users/jonesn/.m2/repository/com/fasterxml/jackson/core/jackson-annotations/2.11.0/jackson-annotations-2.11.0.jar:/Users/jonesn/IdeaProjects/egeria/open-metadata-implementation/common-services/ffdc-services/target/classes:/Users/jonesn/IdeaProjects/egeria/open-metadata-implementation/adapters/open-connectors/rest-client-connectors/rest-client-connectors-api/target/classes:/Users/jonesn/IdeaProjects/egeria/open-metadata-implementation/adapters/open-connectors/rest-client-connectors/rest-client-factory/target/classes:/Users/jonesn/IdeaProjects/egeria/open-metadata-implementation/adapters/open-connectors/rest-client-connectors/spring-rest-client-connector/target/classes:/Users/jonesn/.m2/repository/org/springframework/spring-web/5.2.7.RELEASE/spring-web-5.2.7.RELEASE.jar:/Users/jonesn/.m2/repository/org/springframework/spring-beans/5.2.7.RELEASE/spring-beans-5.2.7.RELEASE.jar:/Users/jonesn/.m2/repository/org/springframework/spring-core/5.2.7.RELEASE/spring-core-5.2.7.RELEASE.jar:/Users/jonesn/.m2/repository/org/springframework/spring-jcl/5.2.7.RELEASE/spring-jcl-5.2.7.RELEASE.jar:/Users/jonesn/.m2/repository/org/codehaus/plexus/plexus-utils/3.3.0/plexus-utils-3.3.0.jar:/Users/jonesn/.m2/repository/org/apache/commons/commons-lang3/3.10/commons-lang3-3.10.jar:/Users/jonesn/.m2/repository/org/slf4j/slf4j-api/1.7.30/slf4j-api-1.7.30.jar:/Users/jonesn/IdeaProjects/egeria/open-metadata-implementation/frameworks/audit-log-framework/target/classes:/Users/jonesn/IdeaProjects/egeria/open-metadata-implementation/access-services/subject-area/subject-area-client/target/classes org.odpi.openmetadata.accessservices.subjectarea.fvt.RunAllFVT
Enter a URL. Press enter to get the default (https://localhost:9443.)):
SLF4J: Failed to load class "org.slf4j.impl.StaticLoggerBinder".
SLF4J: Defaulting to no-operation (NOP) logger implementation
SLF4J: See http://www.slf4j.org/codes.html#StaticLoggerBinder for further details.


Exception in thread "main" java.lang.IllegalArgumentException: can't parse argument number: "timestamp":"2020-06-25T12:03:00.238+00:00"
	at java.base/java.text.MessageFormat.makeFormat(MessageFormat.java:1454)
	at java.base/java.text.MessageFormat.applyPattern(MessageFormat.java:492)
	at java.base/java.text.MessageFormat.<init>(MessageFormat.java:371)
	at org.odpi.openmetadata.frameworks.auditlog.MessageFormatter.formatMessage(MessageFormatter.java:38)
	at org.odpi.openmetadata.frameworks.auditlog.MessageFormatter.getFormattedMessage(MessageFormatter.java:92)
	at org.odpi.openmetadata.frameworks.connectors.ffdc.OCFCheckedExceptionBase.<init>(OCFCheckedExceptionBase.java:75)
	at org.odpi.openmetadata.frameworks.connectors.ffdc.OCFCheckedExceptionBase.<init>(OCFCheckedExceptionBase.java:57)
	at org.odpi.openmetadata.commonservices.ffdc.exceptions.OMAGCheckedExceptionBase.<init>(OMAGCheckedExceptionBase.java:29)
	at org.odpi.openmetadata.accessservices.subjectarea.ffdc.exceptions.SubjectAreaCheckedException.<init>(SubjectAreaCheckedException.java:32)
	at org.odpi.openmetadata.accessservices.subjectarea.ffdc.exceptions.PropertyServerException.<init>(PropertyServerException.java:23)
	at org.odpi.openmetadata.accessservices.subjectarea.client.SubjectAreaBaseImpl.findRESTCall(SubjectAreaBaseImpl.java:260)
	at org.odpi.openmetadata.accessservices.subjectarea.client.SubjectAreaGlossaryImpl.findGlossary(SubjectAreaGlossaryImpl.java:132)
	at org.odpi.openmetadata.accessservices.subjectarea.fvt.GlossaryFVT.findGlossaries(GlossaryFVT.java:210)
	at org.odpi.openmetadata.accessservices.subjectarea.fvt.GlossaryFVT.run(GlossaryFVT.java:61)
	at org.odpi.openmetadata.accessservices.subjectarea.fvt.GlossaryFVT.runWith2Servers(GlossaryFVT.java:35)
	at org.odpi.openmetadata.accessservices.subjectarea.fvt.RunAllFVT.main(RunAllFVT.java:23)
Caused by: java.lang.NumberFormatException: For input string: ""timestamp":"2020-06-25T12:03:00.238+00:00""
	at java.base/java.lang.NumberFormatException.forInputString(NumberFormatException.java:68)
	at java.base/java.lang.Integer.parseInt(Integer.java:638)
	at java.base/java.lang.Integer.parseInt(Integer.java:770)
	at java.base/java.text.MessageFormat.makeFormat(MessageFormat.java:1452)
	... 15 more

Process finished with exit code 1
```

Note - this was run from intellij (I think explains the slf4j warning as no binding - so can ignore that)

I conclude we have a breakage in subject area (FYI @immmus also)

**Comment by immmus** [comment]
Hi @planetf1 
In my new version on https://github.com/odpi/egeria/pull/3245
I don't got this exception.

**Comment by planetf1** [comment]
@immmus thanks.. I'll finish some testing and then will retry with your commits applied to the 2.0 branch - thanks for pointing that out.

**Comment by planetf1** [comment]
The exception is resolved after merging with #3245 - in fact it's likely in exception management, but I think we should go with this fix (master too)

The test still failed, the regression vs 1.8 being caused by #3123

At the core of the issue is that the fvt is semi-hardcoded. It does prompt for input but this is less well suited to an automated environment (but not impossible of course). During the 1.8 cycle I ensured the tests all worked against a standard 'coco' installation as that seems to be our 'demo/test' environment, and we are using coco pharma persona in the test

I don't think it's a big issue for 2.0, it just caused my tests to regress given defaults. 

When (!) we run tests automatically as part of a build process we'll have a 'pre' stage which will bring up some platforms, and for most tests server(s). For FVT we probably would want to stick to single server/non cohort tests - and we should agree on a single name for the server. Anything that needs a cohort would be done as part of a more complex test environment, alongside kafka, multiple platforms etc . Both need more dev work

I would err on at least keeping the current tests and samples working against coco as that is a coherent well described environment, and so would suggest we go back to the second platform (:9444) for the test, but accept it's open to discussion.

**Comment by planetf1** [comment]
This is now being persued adequately in #3245 and the result will be recorded in #3254 as well as work to avoid the confusion and identify errors earlier in #426 so will close this issue
